### DISCUSSION TITLE
gh-pages部署博客，资源文件访问存在项目名重复问题

### DISCUSSION THREAD

**Discussion by Yangly0** [HEIGE-PCloud/DoIt#509] [Q&A]
# 问题

当我用gh-pages部署项目的时候网站存在问题：我设置的`baseURL = "https://xxx.github.io/blog"`，但是
当我去访问文章的时候他访问的是`https://xxx.github.io/blog/blog/posts/`存在两个blog变量名。

![image](https://user-images.githubusercontent.com/43691492/156926629-af800474-7b2d-4865-ad1f-19050471fc2f.png)

![image](https://user-images.githubusercontent.com/43691492/156926644-34d36543-bb9f-44ee-b913-c1a2a88debdd.png)

# 环境
- wind10 
- hugo .093
- chrome

**Comment by cs-mshah** [comment]
Yes, even I am facing this issue of deployment using github-pages. In my case I updated the gh-pages workflow with what is given on the hugo documentation and I see that the hosted site has different routes. Also, it doesn't follow the config and several things are missing. I tried to serve the `./public` folder locally using a local server and it works perfectly fine keeping the routes and config as expected and shown in `hugo serve`. I also symlinked the used config name of `DoIt` to `production` so that the same config is used. But this didn't seem to help.
**Served site:**
![image](https://github.com/HEIGE-PCloud/DoIt/assets/56499208/ad4b8649-7735-4043-8585-5d601490a400 "served site")

**Local site:**
![image](https://github.com/HEIGE-PCloud/DoIt/assets/56499208/29794c33-5c63-4f75-bb1e-a0e24a5f958d)
As you can see, the profile photo and headings are completely missing.

some routes here:
deployed site: `https://cs-mshah.github.io/posts/cs161-computer-security-notes/cs161_computer_security_notes/`
locally hosted: `http://localhost:8787/cs161_computer_security_notes/`

Although the file has been organised in a folder under posts, locally the folder route doesn't get created. I did this just to organise things and found it to work, but on hosting, the folder route is getting created too. But the serving the `./public` folder shows it correctly without creating the folder route for `cs161-computer-security-notes`.

Edit: I changed the environment parameter in the github actions workflow to use the one that I am using `DoIt` instead of symlinking it to production. This works and the site is as expected.

**Reply by cs-mshah** [reply]
But this disables commenting, analytics, so I tried to symlink the `DoIt` config folder to `_default` but it doesn't seem to respect symlinks. 
So finally I think the best way is to just add all config to `_default`. Works!
